File: Selenium/Activities/Activity18.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity18 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();

        // Suppressing browser log
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "NULL");
        // Initialize firefox driver
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));

        // Open a page
        driver.get("https://training-support.net/selenium/selects");

        // Page title
        System.out.println("Title of page is: " + driver.getTitle());


        WebElement dropdown = driver.findElement(By.id("multi-select"));
        Select select = new Select(dropdown);

        // Select the "JavaScript" option
        select.selectByVisibleText("Javascript");
        System.out.println("Text selected: " + select.getFirstSelectedOption().getText());

        // Select the 4th, 5th and 6th options
        select.selectByIndex(5);
        select.selectByIndex(6);
        select.selectByIndex(7);

        // Select the "NodeJS" option
        select.selectByValue("node");
        System.out.println("Text selected: " + select.getFirstSelectedOption().getText());

        // Deselect the 5th option using index.
        select.deselectByIndex(6);

        driver.quit();
    }
}

File: Selenium/Activities/Activity18.py
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the link in browser
    driver.get("https://training-support.net/selenium/selects")

    # Print page title
    print("Page Title is: " + driver.title)

    dropdown = driver.find_element(By.ID, "multi-select")
    dropdown_object = Select(dropdown)

    # Select the "JavaScript" option
    dropdown_object.select_by_visible_text("Javascript")
    print("Text selected: " + dropdown_object.first_selected_option.text)

    # Select the 4th, 5th and 6th options
    dropdown_object.select_by_index(5)
    dropdown_object.select_by_index(6)
    dropdown_object.select_by_index(7)

    # Select the "NodeJS" option
    dropdown_object.select_by_value("node")
    print("Text selected: " + dropdown_object.first_selected_option.text)

    # Deselect the 5th option using index.
    dropdown_object.deselect_by_index(6)

    driver.quit()